Sources/SourceKitLSP/Swift/VariableTypeInfo.swift

```diff
@@ -13,6 +13,35 @@
 import Dispatch
 import LanguageServerProtocol
 import SourceKitD
+import SwiftSyntax
+
+fileprivate extension TokenSyntax {
+  /// Returns `false` if it is known that this token can’t be followed by a type
+  /// annotation.
+  var canBeFollowedByTypeAnnotation: Bool {
+    var node = Syntax(self)
+    LOOP: while let parent = node.parent {
+      switch parent.kind {
+      case .caseItem, .closureParam:
```

Comment:
I was going to ask if this handles the case that the param is in `()`. But apparently it does, because `closureParam` is only the non-signature case. `x in` is a `ClosureParamListSyntax` where as `(x) in` is a `ClosureParameterClauseSyntax`. The former has `ClosureParam`s and the latter `ClosureParameter`s 😅.

Not something that needs fixing here, but worth knowing. I don't have a great name for the former case 🤔.